lib/pf/IPv4.pm

```diff
@@ -0,0 +1,121 @@
+package pf::IPv4;
+
+=head1 NAME
+
+pf::IPv4
+
+=cut
+
+=head1 DESCRIPTION
+
+Object class for handling / managing IPv4 addresses
+
+=cut
+
+use strict;
+use warnings;
+
+use Moose;
+extends 'pf::IP';
+
+# External libs
+use Net::IP;
+
+# Internal libs
+use pf::log;
+
+
+has 'ipv4'  => (is => 'rw');
+
+
+around BUILDARGS => sub {
+    my $orig = shift;
+    my $class = shift;
+
+    my ( $maybe_ip, $maybe_cidr ) = split(m!/!, $_[0]);
+
+    return undef unless ( Net::IP::ip_is_ipv4($maybe_ip) );
+
+    $maybe_cidr = mask2cidr($maybe_cidr) if ( defined($maybe_cidr) && ($maybe_cidr !~ /^\d{1,2}$/) );
+    undef($maybe_cidr) if ( defined($maybe_cidr) && ( ($maybe_cidr < 0) || ($maybe_cidr > 32) ) );
+
+    return $class->$orig (
+        'type'          => 'ipv4',
+        'ipv4'          => $maybe_ip,
+        'prefixLength'  => $maybe_cidr,
+        'normalizedIP'  => Net::IP::ip_expand_address($maybe_ip, 4),
+    );
+};
+
+
+sub is_valid {
+    my $self = shift;
+    if ( ref($self) ne __PACKAGE__ ) {
```

Comment:
Could you add a comment on what this does

Read it a few times and still not sure on what that does